src/app/shared/modal/modal.component.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.css'],
+})
+export class ModalComponent implements OnInit, OnDestroy {
+  @Output() dismiss = new EventEmitter();
+
+  constructor(private el: ElementRef) {}
+
+  ngOnInit(): void {
+    document.body.appendChild(this.el.nativeElement);
+  }
+
+  ngOnDestroy(): void {
+    this.el.nativeElement.remove();
```

Comment:
DOM manipulation like this should be generally avoided. For modal components both Angular Material and Bootstrap provides these components.

Reply:
@Durgaprasad-S404 Were you able to integrate Angular Material or Bootstrap?

Reply:
I didn't integrate it in this project as all the design files will have to be changed. I created a separate one to test out the Material dialog

Reply:
If I understand correctly only the UI and some logic of the modal component will need to be adjusted for this project. Most other files will be left unchanged.

Reply:
Correct. But all UI will have to be done again

Reply:
Yes and that shouldn't be an excuse to not do it. Please take a look [here](https://angular.io/api/core/ElementRef#security-risk) regarding the security risks of using elementRef directly. Doing the UI for a single modal component once again will not take much time.

Reply:
Installing the Material library messed up the UI of every component and not just the UI for the modal. I have implemented the modal using Bootstrap instead.